lua/octo/workflow_runs.lua

```diff
@@ -0,0 +1,701 @@
+        end
+      end
+    else
+      node.expanded = false
+    end
+    M.refresh()
+  end,
+}
+
+local fields =
+  "conclusion,createdAt,databaseId,displayTitle,event,headBranch,headSha,jobs,name,number,startedAt,status,updatedAt,url,workflowDatabaseId,workflowName"
+
+local function get_job_status(status, conclusion)
+  if status == "queued" then
+    return icons.skipped
+  elseif status == "in_progress" then
+    return icons.in_progress
+  elseif conclusion == "success" then
+    return ""
+  elseif conclusion == "failure" then
+    return icons.failed
+  elseif conclusion == "skipped" then
+    return icons.skipped
+  elseif conclusion == "cancelled" then
+    return icons.cancelled
+  else
+    return "❓"
+  end
+end
+
+local function get_step_status(status, conclusion)
+  if status == "pending" then
+    return icons.pending
+  elseif status == "in_progress" then
+    return icons.in_progress
+  elseif conclusion == "success" then
+    return ""
+  elseif conclusion == "failure" then
+    return icons.failed
+  elseif conclusion == "skipped" then
+    return icons.skipped
+  elseif conclusion == "cancelled" then
+    return icons.cancelled
+  else
+    return "❓"
+  end
+end
+
+local function get_workflow_status(status, conclusion)
+  if status == "queued" then
+    return icons.pending
+  elseif status == "in_progress" then
+    return icons.in_progress
+  elseif conclusion == "success" then
+    return icons.succeeded
+  elseif conclusion == "failure" then
+    return icons.failed
+  elseif conclusion == "skipped" then
+    return icons.skipped
+  else
+    return "❓"
+  end
+end
+
+---@type LineDef
+local separator = {
+  value = "",
+  highlight = nil,
+  id = "",
+  type = "separator",
+}
+
+local function get_workflow_header()
+  ---@type WorkflowRun
+  local details = M.current_wf
+  ---@type LineDef[]
+  local lines = {}
+  table.insert(lines, {
+    value = string.format("%s %s", details.displayTitle, get_workflow_status(details.status, details.conclusion)),
+    highlight = "Question",
+  })
+
+  table.insert(lines, separator)
+
+  table.insert(lines, { value = string.format("Branch: %s", details.headBranch), highlight = "Directory" })
+  table.insert(lines, { value = string.format("Event: %s", details.event), highlight = "Directory" })
+
+  if #details.conclusion > 0 then
+    table.insert(
+      lines,
+      { value = string.format("Finished: %s", utils.format_date(details.updatedAt)), highlight = "Directory" }
+    )
+  elseif #details.startedAt > 0 then
+    table.insert(
+      lines,
+      { value = string.format("Started: %s", utils.format_date(details.startedAt)), highlight = "Directory" }
+    )
+  end
+
+  table.insert(lines, separator)
+  return lines
+end
+
+local function update_job_details(id)
+  ---@type WorkflowRun
+  local job_details = {}
+  if M.wf_cache[id] ~= nil then
+    M.refresh()
+    return
+  end
+  local cmd = string.format("gh run view %s --json %s", id, fields)
```

Comment:
Possible to use existing code here? 

Might look like this:

```lua
gh = require "octo.gh"

gh.run.view {
  id,
  json=fields, 
  opts = {
    cb = function(output, stderr)
    end,
  }
}
```

Reply:
You mean like this?
Or did you want me to create the `gh.run.view`?

```
  gh.run {
    args = { "run", "view", id, "--json", fields },
    cb = function(output, stderr)
      if stderr and not utils.is_blank(stderr) then
        vim.api.nvim_err_writeln(stderr)
        octo_error("Failed to get workflow run for " .. id)
      elseif output then
        job_details = vim.fn.json_decode(output)
        M.wf_cache[id] = job_details
        M.current_wf = job_details
        M.tree = generate_workflow_tree(job_details)
        M.refresh()
      end
    end,
  }


```

Reply:
Arbitrary CLI commands are already supported with this logic: 

https://github.com/pwntester/octo.nvim/blob/c689dd516a60665958af9e4195d3bc4a9d5c573b/lua/octo/gh/init.lua?plain=1#L329-L367 

So no need to write it. Just use what I wrote and fill in the callback